src/main/java/org/mdbenefits/app/submission/actions/PreCheckPrograms.java

```diff
@@ -0,0 +1,28 @@
+package org.mdbenefits.app.submission.actions;
+
+import formflow.library.config.submission.Action;
+import formflow.library.data.Submission;
+import java.util.ArrayList;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/**
+ *
+ */
+@Component
+@Slf4j
+public class PreCheckPrograms implements Action {
+
+    @Override
+    public void run(Submission submission) {
+        ArrayList<String> helpNeeded = (ArrayList<String>) submission.getInputData().get("helpNeeded[]");
+        ArrayList<String> programsToAdd = new ArrayList<>();
+
+        helpNeeded.forEach((String need) -> {
```

Comment:
To do: Confirm that if help needed is an empty list, it won't create any errors.

aka, it will just save programs = []

Reply:
Overall, I think it might be better to do the preselect in the page flow rather than here, in an action.   You can key off the "helpNeeded[]" values to set some to preselect.